### ISSUE TITLE
Does not support searching in nested object since `v0.27` 

### ISSUE THREAD

**Issue by Thearas** [meilisearch/meilisearch#2394]
**Describe the bug**

Meilisearch does not support searching nested object since `v0.27`, but it does work fine on `v0.26.1`. Is there any breaking change I missed?😭

**To Reproduce**

1. Prepare data. `createdBy` is a nested object:
  - The index `tests`:
  
  ```json
  {
      "displayedAttributes": [
          "id",
          "createdBy"
      ],
      "searchableAttributes": ["createdBy"]
  }
  ```

  - The document:
  ```json
  {
    "id": 1,
    "createdBy": {
      "name": "foo"
    }
  }
  ```

2. Now run

  ```bash
  curl 'http://localhost:7700/indexes/tests/search' -H 'Content-Type: application/json' \
  --data-raw '{"q":"foo"}'
  ```

  - For `v0.26.1` ✅:
    ```json
    {
      "hits": [
        {
          "createdBy": {
            "name": "foo"
          },
          "id": 1
        }
      ],
      "nbHits": 1,
      "exhaustiveNbHits": false,
      "query": "foo",
      "limit": 20,
      "offset": 0,
      "processingTimeMs": 0
    }
    ```

  - For `v0.27` ❌:
    No hits.

**Meilisearch version:**

`v0.27`

**Comment by curquiza** [comment]
Hello @Thearas 

Thanks a lot for this report, we will release a patch release (v0.27.1) of Meilisearch as soon as possible!

**Comment by irevoire** [comment]
Hey @Thearas, I can reproduce the bug thanks for the report!
It looks like the bug is related to the `searchableAttributes` not going deep into your structure.

In the meantime, if you want to try the v0.27.0 you can update the `searchableAttributes` to: `createdBy.name`. That solved the issue for me.

I'll try to fix this bug ASAP!

**Comment by irevoire** [comment]
Hey @Thearas, the bug has been fixed in https://github.com/meilisearch/milli/pull/528
Now we need to wait until we release a new version of milli, then I'll integrate it in meilisearch and we'll be able to release a v0.27.1. It'll probably come by tomorrow, maybe before if the CIs are fast.

Then you'll need to reindex your documents or settings somehow. You can either update your `searchableAttributes` or make an add documents operation :+1:

**Comment by Thearas** [comment]
Wow, that's very fast 👍 @irevoire 
The solution works for me, so this bug did not block my work:
> In the meantime, if you want to try the v0.27.0 you can update the searchableAttributes to: createdBy.name. That solved the issue for me.

THK!

**Comment by curquiza** [comment]
Fixed by #2387, will be released in v0.27.1
